cql3/statements/select_statement.cc

```diff
@@ -2012,6 +2012,7 @@ std::unique_ptr<prepared_statement> select_statement::prepare(data_dictionary::d
 
     if (_parameters->is_distinct()) {
         validate_distinct_selection(*schema, *selection, *restrictions);
+        validate_distinct_syntax_constraints();
```

Comment:
Super annoying nitpick: Cassandra runs this validation _before_ doing the analogous work in `validate_distinct_selection` (called `validateDistinctSelection` there). Let's switch the order of the lines to make the logic resemble Cassandra more closely, and to make the error-generating behavior more similar to C*, too.